src/application/listeners/projectEventListeners.ts

```diff
@@ -1,81 +1,99 @@
     }
+
+    await newProject(projectInfo.id, projectInfo.path, projectInfo.name);
+    createFile();
+    notifyInfo('New project created with success');
+    persistProjectIdInSettings(projectInfo.id);
   };
 }
 
 export function useFileProjectNewSampleListener() {
   const sampleProject = useSetAtom(newSampleProjectAtom);
+  const createFile = useSetAtom(createFileAtom);
   return async () => {
-    const newProjectPath = `${await getNewProjectsBaseDir()}RefStudio Sample`;
-    await sampleProject(newProjectPath);
-    persistProjectDirInSettings(newProjectPath);
-    notifyInfo('New project created at ' + newProjectPath);
+    const projects = await readAllProjects();
+    let project = projects.find((p) => p.name === SAMPLE_PROJECT_NAME);
+    if (!project) {
+      project = await createRemoteProject('RefStudio Sample');
+    }
+    await sampleProject(project.id, project.name, project.path);
+    createFile();
+    notifyInfo('Sample project opened with success');
+    persistProjectIdInSettings(project.id);
   };
 }
 
 export function useFileProjectOpenListener() {
   const openProject = useSetAtom(openProjectAtom);
-  const openWebProject = useSetAtom(openWebProjectAtom);
 
   return async () => {
+    let projectId: string;
     if (import.meta.env.VITE_IS_WEB) {
       // TODO: Open a custom dialog with all projects to select one
       // For now, opens the first project found
-      const projectInfo = await readOneProjectFromWeb();
-      return openWebProject(projectInfo.id, projectInfo.path, projectInfo.name);
-    }
-
-    const selectedPath = await open({
-      directory: true,
-      multiple: false,
-      defaultPath: await getNewProjectsBaseDir(),
-      title: 'Open RefStudio project',
-    });
+      const projects = await readAllProjects();
+      projectId = projects[0].id;
+    } else {
+      const selectedPath = await open({
+        directory: true,
+        multiple: false,
+        defaultPath: await getNewProjectsBaseDir(),
+        title: 'Open RefStudio project',
+      });
 
-    if (typeof selectedPath === 'string') {
-      await openProject(selectedPath);
-      persistProjectDirInSettings(selectedPath);
-      notifyInfo('New project open at ' + selectedPath);
+      if (typeof selectedPath === 'string') {
+        const projects = await readAllProjects();
+        const projectWithSamePath = projects.find((project) => project.path === selectedPath);
+        if (!projectWithSamePath) {
+          notifyWarning('You need to open a project that was previously created with RefStudio.');
+          return;
+        }
+        projectId = projectWithSamePath.id;
+      } else {
+        notifyInfo('User canceled operation to open new project');
+        return;
+      }
     }
+
+    const projectToOpen = await readProjectById(projectId);
+    await openProject(projectId, projectToOpen.path, projectToOpen.name);
+    persistProjectIdInSettings(projectId);
```

Comment:
Could we name this something like `persistCurrentProject` or `persistActiveProject`? Is there a way we could avoid having to call it in so many places? Every time you open a project, we should persist it.

Reply:
We can call it diferently but I think we can't avoid calling in these 3 places (new, open, sample).